src/Axe/Managers/CompilerManager.cs

```diff
@@ -77,20 +96,41 @@ public CompilerManager(AxeDbContext context) : base(context)
             }
         }
 
-        public async Task Update(CodeBlockTaskVm model)
+        public async Task Create(CodeBlockCreateVm model)
+        {
+            try
+            {
+                CodeBlock codeBlock = new CodeBlock();
+                codeBlock.Task = model.Task;
+                codeBlock.TestCases = model.TestCases;
+                codeBlock.OutputType = model.OutputType;
+                codeBlock.VerificationCode = GenerateVerificationCodeCSharp(model);
+                codeBlock.Technology = await context.Technology.FirstAsync(t => t.Id == model.SelectedTechnologyId);
+                context.CodeBlock.Add(codeBlock);
+                await context.SaveChangesAsync();
+            }
+            catch (Exception exception)
+            {
+                throw exception;
+            }
```

Comment:
and the purpose of exception rethrow is ... ?

Reply:
It rethrows any exception to a higher level (in our case - to compiler where this exception can used to show error message for user)

Reply:
`throw exception` changes original StackTrace. is it intended? if not - higher level will just get exception if we don't have `try {} catch {}` here

Reply:
aa1caec